wagtail/admin/forms/account.py

```diff
@@ -82,7 +83,16 @@ def __init__(self, *args, **kwargs):
     )
 
     current_time_zone = forms.ChoiceField(
-        required=False, choices=_get_time_zone_choices, label=_("Current time zone")
+        required=False,
+        choices=_get_time_zone_choices,
+        label=_("Current time zone"),
+        widget=forms.Select(
+            attrs={
+                "data-controller": "w-init w-locale",
+                "data-action": "w-init:ready->w-locale#localizeTimeZoneOptions",
```

Comment:
Not critical but just a reminder that if any events with the same name dispatch from within the select, this will bubble up and trigger this.

It's very unlikely within a select field, however, it's probably recommended to add `:self` - so that only events dispatched on that specific DOM element will activate the action. https://stimulus.hotwired.dev/reference/actions#options

```
                "data-action": "w-init:ready->w-locale#localizeTimeZoneOptions:self",
```

Alternatively, call this from within the controller's `connect` method.

Again, not likely to cause an issue for this instance of usage, but just flagging for future reference. Also, even if it does get fired additional times, it will just run the same update & not really change the DOM from what I can see.

Reply:
Thank you, that makes sense! I don't recall that option before, I'll keep it in mind for future use.